homeassistant/components/home_connect/select.py

```diff
@@ -28,6 +24,166 @@
 
 _LOGGER = logging.getLogger(__name__)
 
+PROGRAMS_MAP = {
+    bsh_key_to_translation_key(program): program
```

Comment:
I'd invert this map and write it out without the regex parsing. Then create an inverse dict as well from this dict, so we have two constant maps to look up for both programs and translation keys.

Reply:
Why? We already have the function that allows us to get the translation keys from the programs.
Additionally, if we don't use a constant map for getting translation key from the program, the entities will be able to show programs that are not documented at the API documentation (although then they will no be able to use it due to the map we already have....)

Reply:
If we have a map we don't need a function to generate a string.

Reply:
It's better a map than a function?

Reply:
Yes, it's easier to read and faster.

Reply:
Okey then, let's do the double map :)
It's okey if we keep using the function just to generate the maps?

Reply:
Sure. I guess it may be convenient when adding many new keys at the same time.